### ISSUE TITLE
There seems to be something wrong with the sidebar

### ISSUE THREAD

**Issue by dxj728** [nextcloud/android#2243]
### Actual behaviour
When you collect a folder, open the sidebar, select your favorites option, then find your favorite folder in the interface, then click enter the folder, it seems that everything looks normal, but when you return, But found that the previous directory shows all the files, not the original collection of those files, but the sidebar options still remain in the collection interface, I think this is very unreasonable

### Expected behaviour
No matter how you click on the favorites asking price folder, exit should be consistent with the previous interface, showing the collection of files, not all
 
### Steps to reproduce
1. Collect a folder
2. Then click enter the folder in the favorites screen of the sidebar
3. If you go back to the root directory, you will find that all the files are displayed in the root directory, which does not seem reasonable

### Environment data
Android version: android 5.1.1

Device model: Nexus5

Stock or customized system:

Nextcloud app version :v3.0.1

Nextcloud server version: 12.0.0

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Can you do some screenshots for the steps you mentioned as I unfortunately do not understand the problem.

**Comment by dxj728** [comment]
When you click from the sidebar's favorites screen to enter a folder you've collected, and then go back to the previous interface to the root directory, you'll find that the root directory shows all the files, not just the ones you used to collect.

**Comment by dxj728** [comment]
This is the correct collection display interface
![screenshot_1519784142](https://user-images.githubusercontent.com/18740322/36766374-9188090c-1c70-11e8-9354-fe3798d911dd.png)
![screenshot_1519784148](https://user-images.githubusercontent.com/18740322/36766390-a6daf954-1c70-11e8-80c4-e77b50085d78.png)
When you click on the folder of this collection, and then exit the folder, it seems that the collection interface has disappeared and become the display of all files, but this is still under the collection interface, which I think is wrong.

**Comment by dxj728** [comment]
![screenshot_1519784168](https://user-images.githubusercontent.com/18740322/36766596-825baf00-1c71-11e8-8515-ad09cd713fa0.png)

**Comment by dxj728** [comment]
@tobiasKaminsky Hello, I have provided a screenshot, please reconfirm this question.

**Comment by tobiasKaminsky** [comment]
Now I got it, then this will be fixed with #1272
